common/src/main/java/com/tencent/rss/common/util/RssUtils.java

```diff
@@ -135,4 +141,54 @@ public static Roaring64NavigableMap deserializeBitMap(byte[] bytes) throws IOExc
     bitmap.deserialize(dataInputStream);
     return bitmap;
   }
+
+  public static List<ShuffleDataSegment> transIndexDataToSegments(ShuffleIndexResult sir, int readBufferSize) {
+    if (sir == null || sir.isEmpty()) {
+      return Lists.newLinkedList();
+    }
+
+    byte[] indexData = sir.getIndexData();
```

Comment:
indexData may be empty or null？

Reply:
Yes

Reply:
this has been done in the upper if block